npc_relationships.lua

```diff
@@ -0,0 +1,88 @@
+
+local PLUGIN = PLUGIN
+
+PLUGIN.name = "NPC Relationships"
+PLUGIN.author = "dave"
+PLUGIN.description = "Allows certain NPCs to be friendly to certain player factions. Supports VJBase NPCs."
+PLUGIN.license = [[
+Copyright 2021 dave
+Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated documentation files (the "Software"), to deal in the Software without restriction, including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+]]
+
+if SERVER then -- if on server
+
+local zombieNPCs = {
+	"npc_zombie",
+	"npc_vj_nmrih_runfemalez",
+	"npc_vj_nmrih_runmalez",
+	"npc_vj_nmrih_runsoldierz",
+	"npc_vj_nmrih_walkfemalez",
+	"npc_vj_nmrih_walkmalez",
+	"npc_vj_nmrih_walksoldierz",
+	"npc_vj_nmrih_childz",
+
+}
+
+local friendlyFaction = FACTION_ZOMBIES -- change this to the faction you want your npcs to be friendly to
+
+local friendlyClasses = { -- vjbase classes to make friendly if the player is a zombie
+						  -- if not using vjbase, leave as is
+	"CLASS_ZOMBIE",
+
+}
+
+function SetRelationship(ent, ply, relationship) -- function to set relationship
+	ent:AddEntityRelationship(ply, relationship, 99) -- make ent friendly to ply
+	

```

Comment:
Same as above + I don't see why this function should be global and how it can help shortening the code in any way. It should simply be removed.